python3Packages.python-dotenv: 1.0.0 -> 1.0.1

mihaimaruseac opened the pull request:
## Description of changes

Updating this package as the new version was released this January and a package I want to add to nixpkgs depends on the newer version. The new version also supports newer versions of Python.

Release notes: https://github.com/theskumar/python-dotenv/releases/tag/v1.0.1

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

mihaimaruseac committed:
python3Packages.python-dotenv: 1.0.0 -> 1.0.1

Updating this package as the new version was released this January and a
package I want to add to nixpkgs depends on the newer version. The new version
also supports newer versions of Python.

Release notes: https://github.com/theskumar/python-dotenv/releases/tag/v1.0.1

Signed-off-by: Mihai Maruseac <mihai.maruseac@gmail.com>

kirillrdy commented:
greetings !
this should target staging branch

there is a common gotcha that is documented here https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

mihaimaruseac commented:
Thank you for the suggestion! Rebased to staging.

mihaimaruseac commented:
I got the reviewers added by Github when I switched the base branch :( I should have moved the PR to draft before doing the switch and move back to ready state after it was done. Do you want me to close this and reopen?

Regarding commit message, thank you, I've updated it
